Good Coding Practices/src/js/piece.js

```diff
@@ -0,0 +1,75 @@
+// piece object
+const piece = (function () {
+    // let el = null;
+    const init = function (el) {
+        this.el = el;
+        this.setPieceColorByTemp()
+    }
+    const moveDelta = function (dx, dy) {
+        const pos = this.el.getBoundingClientRect();
+        let yPos;
+        let xPos;
+
+        if (dx === 0)
+            yPos = this.movePiece(pos.top, dy, (window.innerHeight - 100));
+        else
+            xPos = this.movePiece(pos.left, dx, (window.innerWidth - 100));
+
+        this.setElePos(yPos, xPos);
+    }
+    const movePiece = function (curPos, ds, size) {
+        let pos = 0;
+        let tmpPos = curPos + ds;
+        if (tmpPos > 0 && tmpPos < size)
+            pos = tmpPos;
+        else if (tmpPos > 0 && tmpPos > size)
+            pos = size;
+        else if (tmpPos < 0 && tmpPos < size)
+            pos = 0;
+
+        return pos;
+    }
+    const resetPieceLoc = function () {
+        const pos = JSON.parse(sessionStorage.getItem("defPos"));
+        this.setElePos(pos.top, pos.left);
+    }
+    const randPieceLoc = function () {
+        const randLocTop = Math.floor(Math.random() * (window.innerHeight - 100));
+        const randLocLeft = Math.floor(Math.random() * (window.innerWidth - 100));
+        this.setElePos(randLocTop, randLocLeft);
+    }
+    const setElePos = function (topPos, leftPos) {
+        this.el.style.top = `${topPos}px`;
+        this.el.style.left = `${leftPos}px`;
+    }
+    const setPieceColorByTemp = function () {
```

Comment:
`Piece` does not need to know there's an API and how it works. It's not its responsibility